Author: Andreas Henriksson <andreas@fatal.se>
License: GNU GPL, see /usr/share/common-licenses/GPL
Description: Fixes crashing on startup due to wrong types
Bug-Debian: http://bugs.debian.org/437330
Last-Update: 2007-11-20
--- a/src/DummyAmanith.py
+++ b/src/DummyAmanith.py
@@ -42,7 +42,9 @@
     pass
 
   def SetViewport(self, x, y, w, h):
-    glViewport(int(x), int(y), int(w), int(h))
+    glw = int(w)
+    glh = int(h)
+    glViewport(int(x), int(y), int(glw), int(glh))
 
   def SetProjection(self, left, right, bottom, top):
     glMatrixMode(GL_PROJECTION)
--- a/src/GameEngine.py
+++ b/src/GameEngine.py
@@ -165,7 +165,9 @@
     geometry = (0, 0, w, h)
     self.svg = SvgContext(geometry)
     self.svg.setRenderingQuality(self.config.get("opengl", "svgquality"))
-    glViewport(int(viewport[0]), int(viewport[1]), int(viewport[2]), int(viewport[3]))
+    glw = int(viewport[2])
+    glh = int(viewport[3])
+    glViewport(int(viewport[0]), int(viewport[1]), int(glh), int(glh))
 
     self.input     = Input()
     self.view      = View(self, geometry)
--- a/src/View.py
+++ b/src/View.py
@@ -136,10 +136,10 @@
 
     viewport = glGetIntegerv(GL_VIEWPORT)
     if normalize:
-      w = viewport[2] - viewport[0]
-      h = viewport[3] - viewport[1]
+      w = int ( viewport[2] - viewport[0] )
+      h = int ( viewport[3] - viewport[1] )
       # aspect ratio correction
-      h *= (float(w) / float(h)) / (4.0 / 3.0)
+      h *= int((float(w) / float(h)) / (4.0 / 3.0))
       viewport = [0, 0, 1, h / w]
   
     if yIsDown:
